frontend/app/routes/_gcweb-app.personal-information.home-address.edit.tsx

```diff
@@ -81,21 +81,13 @@ export async function action({ request }: ActionFunctionArgs) {
       formData: formData as Partial<z.infer<typeof formDataSchema>>,
     });
   }
-  const wsAddressWebService = await getWSAddressService();
-  const validateResultAddress = await wsAddressWebService.validateWSAddress({
-    addressLine: parsedDataResult.data.address,
-    city: parsedDataResult.data.city,
-    country: parsedDataResult.data.country,
-    language: 'EN',
-    postalCode: parsedDataResult.data.postalCode ?? '',
-    province: parsedDataResult.data.province ?? '',
-    geographicScope: 'GS',
-    parseType: 'PT',
-  });
 
   const sessionService = await getSessionService();
   const session = await sessionService.getSession(request);
   session.set('newHomeAddress', parsedDataResult.data);
+
+  const wsAddressWebService = await getWSAddressService();
+  const validateResultAddress = await wsAddressWebService.validateWSAddress(parsedDataResult.data.address, parsedDataResult.data.city, parsedDataResult.data.country, parsedDataResult.data.postalCode ?? '', parsedDataResult.data.province ?? '');
```

Comment:
One concern I have about this is how easy it is to introduce a bug by accidentally sending the args in the wrong order.

When you have a function with `(string, string, string, string, string)` for args, it is more difficult to tell at a glance if you have the args in the correct order.

Reply:
It also gets a bit wonky when some of the args are optional. (I know that's not the case here, just making a point)

Reply:
too bad JS doesn't have keyword arguments

Reply:
Fixed. The function now accepts an object rather than a bunch o' strings.